client/client.go

```diff
@@ -43,51 +46,90 @@ type CloseFunc func(context.Context) error
 // A function is returned to close the client. It is also strongly recommended
 // to call this when the program is exiting or the lekko provider is no longer needed.
 func NewClient(provider Provider) (Client, CloseFunc) {
-	return newClient(provider, nil)
+	return newClient(provider, ClientOptions{})
 }
 
 type ClientOptions struct {
 	// Arbitrary keys and values for rules evaluation that are
 	// injected into the context at startup. Context keys passed
 	// at runtime will override these values in case of conflict.
-	StartupContext map[string]interface{}
+	StartupContext    map[string]interface{}
+	EnableOTelTracing bool
 }
 
 func (o ClientOptions) NewClient(provider Provider) (Client, CloseFunc) {
-	return newClient(provider, o.StartupContext)
+	return newClient(provider, o)
 }
 
 type client struct {
-	provider       Provider
-	startupContext map[string]interface{}
+	provider          Provider
+	startupContext    map[string]interface{}
+	enableOTelTracing bool
 }
 
-func newClient(provider Provider, startupCtx map[string]interface{}) (*client, CloseFunc) {
-	return &client{provider: provider, startupContext: startupCtx}, provider.Close
+func newClient(provider Provider, options ClientOptions) (*client, CloseFunc) {
+	return &client{
+		provider:          provider,
+		startupContext:    options.StartupContext,
+		enableOTelTracing: options.EnableOTelTracing,
+	}, provider.Close
+}
+
+func (c *client) addTracingEvent(ctx context.Context, key, stringValue, version string) {
+	if !c.enableOTelTracing {
+		return
+	}
+	span := trace.SpanFromContext(ctx)
+	attributes := []attribute.KeyValue{
+		attribute.String("config.key", key),
+	}
+	if len(stringValue) > 0 {
+		attributes = append(attributes, attribute.String("config.string_value", stringValue))
+	}
+	if len(version) > 0 {
+		attributes = append(attributes, attribute.String("config.version", version))
```

Comment:
I started with `version` here, but then I named it `LastUpdateCommitSHA` in our protos. I'm not sure which one is better, any suggestions?

Reply:
i think the latter is more descriptive and works well